### PR TITLE
Feature/home screen

### PR DESCRIPTION
**Pull request by marina-ivanovic**
Changed designs of event/offering cards, added location and date attributes to the Event class, changed from tabs for displaying the home page to segmented button.

### CODE REVIEW

File: EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopEventsFragment.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        // Initialize the adapter and set it to RecyclerView
+        adapter = new EventListAdapter(eventList);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        recyclerView.setAdapter(adapter);
+
+        return rootView;
+    }
+
+    // Method to simulate getting top event data
+    private List<Event> getTopEvents() {
```
Comment (milicaradicc) [inline EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopEventsFragment.java:43]: Add some error handling or empty state management. For example, if the list of events is empty, you can display a "No Events Available" message or some fallback UI

File: EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopOfferingsFragment.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_top_offerings, container, false);
+
+        // Initialize RecyclerView
+        recyclerView = rootView.findViewById(R.id.topOfferingsRecyclerView);
+
+        // Initialize offeringList (you will replace this with actual data fetching logic)
+        offeringList = getOfferingList();  // Ensure this method provides data
```
Comment (milicaradicc) [inline EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopOfferingsFragment.java:35]: Add some error handling or empty state management

**Review by milicaradicc — CHANGES_REQUESTED** [review]

File: EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopOfferingsFragment.java (hunk 2 above)
Comment (milicaradicc) [reply EventPlanner/app/src/main/java/m3/eventplanner/fragments/TopOfferingsFragment.java:35]: Design looks great,  implementing an empty state is necessary. Add error handling and loading states for a better user experience.

**Comment by milicaradicc** [comment]
Design looks great, implementing an empty state is necessary. Add error handling and loading states for a better user experience.

**Comment by marina-ivanovic** [comment]
Changes made:
- Made a display text for when the list of events/offerings is empty.
- Combined all of the xml and Java files for display of cards (allEvents/topEvents/allOfferings/topOfferings fragments and xml files) into one fragment and created the logic for displaying and hiding components depending on the tab the user clicked on.
- Changed the search bar from custom made one, to material design search bar
- Renamed the filter components, but the components themselves will be changed on another branch

- There is no longer need to separate the recycler view from the component it's getting used in, if you want to reuse the component, there is no need to make another xml, just make the recycler view for that list. The logic for putting the the cards in the recycler view is in the home fragment, and the adapters for events/offerings can be used for binding the data to the cards.

**Review by milicaradicc — APPROVED** [review]

**Pull request merged**